src/functions/runtimeConfigExport.ts

```diff
@@ -0,0 +1,222 @@
+import * as fs from "fs";
+import * as path from "path";
+
+import * as clc from "cli-color";
+
+import * as env from "./env";
+import * as functionsConfig from "../functionsConfig";
+import { FirebaseError } from "../error";
+import { logger } from "../logger";
+import { getProjectId } from "../projectUtils";
+import { loadRC } from "../rc";
+import { logWarning } from "../utils";
+
+export interface ProjectConfigInfo {
+  projectId: string;
+  alias?: string;
+  config?: Record<string, unknown>;
+  envs?: EnvMap[];
+}
+
+export interface EnvMap {
+  origKey: string;
+  newKey: string;
+  value: string;
+  err?: string;
+}
+
+export interface ConfigToEnvResult {
+  success: EnvMap[];
+  errors: Required<EnvMap>[];
+}
+
+/**
+ * Find all projects (and its alias) associated with the current directory.
+ */
+export function getProjectInfos(options: {
+  project?: string;
+  projectId?: string;
+  cwd?: string;
+}): ProjectConfigInfo[] {
+  const result: Record<string, string> = {};
+
+  const rc = loadRC(options);
+  if (rc.projects) {
+    for (const [alias, projectId] of Object.entries(rc.projects)) {
+      if (Object.keys(result).includes(projectId)) {
+        logWarning(
+          `Multiple aliases found for ${clc.bold(projectId)}. ` +
+            `Preferring alias (${clc.bold(result[projectId])}) over (${clc.bold(alias)}).`
+        );
+        continue;
+      }
+      result[projectId] = alias;
+    }
+  }
+
+  const projectId = getProjectId(options);
+  if (projectId && !Object.keys(result).includes(projectId)) {
+    result[projectId] = projectId;
+  }
+
+  return Object.entries(result).map(([k, v]) => {
+    const result: ProjectConfigInfo = { projectId: k };
+    if (k !== v) {
+      result.alias = v;
+    }
+    return result;
+  });
+}
+
+/**
+ * Fetch and fill in runtime config for each project.
+ */
+export async function hydrateConfigs(pInfos: ProjectConfigInfo[]): Promise<void> {
+  for (const pInfo of pInfos) {
```

Comment:
Let's do this in parallel. Even something silly works like:

```
await Promise.all(pInfos.map((async (info) => {
  try {
    pInfo.config = await functionsConfig.materliazeAll(pInfo.projectId);
  } catch (err) {
    logger.debug(...);
  }
})()));
```

Should do the trick (note, i made your mapped function async and invoked it immediately to make the whole iteration a promise)